### ISSUE TITLE
Add a lightweight AverageMeter / metric tracking utility to torch.utils

### ISSUE THREAD

**Issue by powerofaisinstudy-debug** [pytorch/pytorch#194997]
### 🚀 The feature, motivation and pitch

When writing training loops in native PyTorch, it's super common to track losses  or metrics by appending them to lists or dicts. But if someone accidentally passes a raw tensor without calling `.detach().item()`, it causes a silent graph-retention memory leak that quietly eats up VRAM until an OOM crash happens.

While there are huge metrics libraries out there, it would be really nice to have a tiny, built-in utility inside `torch.utils` (like an `AverageMeter`) that handles tensor detachment and running averages automatically. Every developer ends up re-implementing this exact same helper in their repos anyway.

#Proposed Api

`loss_meter = AverageMeter(name="Loss", fmt=":.4f")`

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by malfet** [comment]
Sounds like an interesting idea. Do not hesitate implement it out of tree, find some people wanting to try it and if feedback is positive do not hesitate to open another feature request to integrate it
